site/src/components/pages/hub/BlockDataContainer.tsx

```diff
@@ -264,6 +307,20 @@ export const BlockDataContainer: VoidFunctionComponent<
               </Box>
             </Box>
           )}
+
+          <Snackbar
+            open={alertSnackBarOpen}
+            autoHideDuration={4000}
+            anchorOrigin={{
+              horizontal: "right",
+              vertical: "bottom",
+            }}
+            onClose={() => setAlertSnackBarOpen(false)}
+          >
+            <Alert severity="warning">
+              Please fix the errors in the <b>Data Source</b> before proceeding.{" "}
```

Comment:
Display an alert to fix block errors in case the user tries to switch variants with a broken data source.

This is because we have to parse the existing text while applying the new properties of a variant to the `text` state.